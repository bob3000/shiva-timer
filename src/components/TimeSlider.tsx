import * as React from 'react';
import { Slider, StyleSheet, TextStyle, View, ViewStyle } from 'react-native';
import Timer from '../components/Timer';

export interface ITimeSliderProps {
  disabled?: boolean;
  intervals: number[];
  onSlidingComplete?: () => void;
  onValueChange?: (value: number) => void;
  value: number;
}

export const TimeSlider: React.FunctionComponent<ITimeSliderProps> = (
  props,
) => {
  return (
    <View style={styles.sliderContainer}>
      <Slider
        disabled={props.disabled}
        maximumTrackTintColor={'#000000'}
        maximumValue={props.intervals.length - 1}
        minimumTrackTintColor={'#000000'}
        minimumValue={0}
        onSlidingComplete={props.onSlidingComplete}
        onValueChange={(value: number) => {
          if (props.onValueChange) {
            props.onValueChange(props.intervals[Math.trunc(value)]);
          }
        }}
        style={styles.slider}
        thumbTintColor={'#000000'}
        value={props.intervals.indexOf(props.value)}
      />
      <Timer
        digitSpace={22}
        displayTimeStyle={styles.scaleText}
        displayTime={props.value}
        isEditMode={false}
      />
    </View>
  );
};

export interface ITimeSliderStyle {
  scaleText: TextStyle;
  slider: TextStyle;
  sliderContainer: ViewStyle;
}

const styles = StyleSheet.create<ITimeSliderStyle>({
  scaleText: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  slider: {
    width: 200,
  },
  sliderContainer: {
    flexDirection: 'row',
    padding: 15,
  },
});

export default TimeSlider;
