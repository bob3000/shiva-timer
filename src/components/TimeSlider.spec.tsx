import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TimeSlider from './TimeSlider';

describe('TimeSlider', () => {
  it('snapshot test', () => {
    const tree = renderer
      .create(<TimeSlider intervals={[1, 2, 4]} value={4} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
