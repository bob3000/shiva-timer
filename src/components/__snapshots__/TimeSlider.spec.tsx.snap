// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSlider snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "padding": 15,
      }
    }
  >
    <RCTSlider
      disabled={false}
      enabled={true}
      maximumTrackTintColor="#000000"
      maximumValue={2}
      minimumTrackTintColor="#000000"
      minimumValue={0}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onValueChange={[Function]}
      step={0}
      style={
        Array [
          Object {
            "height": 40,
          },
          Object {
            "width": 200,
          },
        ]
      }
      thumbTintColor="#000000"
      value={2}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "width": 25,
              },
              Object {
                "textAlign": "right",
              },
            ]
          }
        >
          00
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "bold",
              },
            ]
          }
        >
          :
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "width": 25,
              },
            ]
          }
        >
          04
        </Text>
      </View>
    </View>
  </View>
</View>
`;
